### PR TITLE
Add demo for Caffe2, support caffe2.python.model_helper.ModelHelper, fix bug in #179

### PR DESCRIPTION
**Pull request by ml7**

### CODE REVIEW

**Comment by ml7** [comment]
screen shot 2018-07-05 at 11 24 37 am
screen shot 2018-07-05 at 11 24 45 am

Some screenshots showing the graph visualization functionality is working, as well as the scalar value plotting for both raw numerical values and blob names.

File: tensorboardX/writer.py [hunk 1]
```diff
@@ -450,7 +450,7 @@ def add_audio(self, tag, snd_tensor, global_step=None, sample_rate=44100):
             snd_tensor: :math:`(1, L)`. The values should lie between [-1, 1].
         """
         if self._check_caffe2(snd_tensor):
-            snd_tensor = workspace.FetchBlobl(snd_tensor)
```
Comment (lanpa) [inline tensorboardX/writer.py]: It that a typo? Please let someone familiar with caffe2 to check for such trivial error.

File: tensorboardX/writer.py [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 try:
     from caffe2.proto import caffe2_pb2
-    from caffe2.python import cnn, core, workspace
+    from caffe2.python import cnn, core, model_helper, workspace
```
Comment (lanpa) [inline tensorboardX/writer.py]: Where do you use model_helper ?

File: tensorboardX/writer.py (hunk 1 above)
Comment (ml7) [reply tensorboardX/writer.py]: Yes this was a typo

File: tensorboardX/writer.py (hunk 2 above)
Comment (ml7) [reply tensorboardX/writer.py]: I used it to check if the model is an instance of model_helper.ModelHelper, but I later changed the check so it first checks whether it is a core.Net or caffe2_pb2.NetDef, then if it is neither, it just assumes it is a Caffe2 model. I can delete this?

Comment (lanpa) [reply tensorboardX/writer.py]: Yes, unnecessary import should be removed.

**Pull request merged**